src/pages/billing/List/List.ts

```diff
@@ -0,0 +1,257 @@
+import _ from 'lodash/fp'
+import * as qs from 'qs'
+import { useEffect, useRef, useState } from 'react'
+import { div, h, h2, p, span } from 'react-hyperscript-helpers'
+import Collapse from 'src/components/Collapse'
+import { customSpinnerOverlay } from 'src/components/common'
+import FooterWrapper from 'src/components/FooterWrapper'
+import TopBar from 'src/components/TopBar'
+import { Ajax } from 'src/libs/ajax'
+import * as Auth from 'src/libs/auth'
+import colors from 'src/libs/colors'
+import { reportErrorAndRethrow } from 'src/libs/error'
+import Events from 'src/libs/events'
+import * as Nav from 'src/libs/nav'
+import { useCancellation, useOnMount, useStore } from 'src/libs/react-utils'
+import { authStore } from 'src/libs/state'
+import * as StateHistory from 'src/libs/state-history'
+import * as Style from 'src/libs/style'
+import * as Utils from 'src/libs/utils'
+import { CloudProvider, cloudProviderTypes } from 'src/libs/workspace-utils'
+import { billingRoles } from 'src/pages/billing/Billing'
+import { CreateBillingProjectControl } from 'src/pages/billing/List/CreateBillingProjectControl'
+import { GCPNewBillingProjectModal } from 'src/pages/billing/List/GCPNewBillingProjectModal'
+import { ProjectListItem } from 'src/pages/billing/List/ProjectListItem'
+import { BillingProject } from 'src/pages/billing/models/BillingProject'
+import { GoogleBillingAccount } from 'src/pages/billing/models/GoogleBillingAccount'
+import { AzureBillingProjectWizard } from 'src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AzureBillingProjectWizard'
+import { GCPBillingProjectWizard } from 'src/pages/billing/NewBillingProjectWizard/GCPBillingProjectWizard/GCPBillingProjectWizard'
+import ProjectDetail from 'src/pages/billing/Project'
+
+
+const isCreatingStatus = status => _.includes(status, ['Creating', 'CreatingLandingZone'])
+
+const BillingProjectSubheader = ({ title, children }) => h(Collapse, {
+  title: span({ style: { fontWeight: 'bold' } }, [title]),
+  initialOpenState: true,
+  titleFirst: true,
+  summaryStyle: { padding: '1rem 1rem 1rem 2rem' }
+}, [children])
+
+export const List = ({ queryParams: { selectedName } }) => {
```

Comment:
I'm also new and struggle with the syntax for declaring types. @blakery do you know the answer to this?

FWIW, there is no type definition yet for `queryParams`.

Reply:
Something like:

```ts
interface ListProps {
  queryParams: {
    selectedName: string
  }
}

export const List = (props: ListProps) => {
  ...
```

should work.

Reply:
Thanks @nawatts . so there is no way to declare a type definition like that without the interface? Or is it just that we generally prefer "Props" interfaces (which I've been using elsewhere, so I would understand that).

Reply:
Alternatively, if query parameters are coming directly from the URL, then the type should probably reflect that selectedName isn't guaranteed to be present (or might be an array or object). Something like:

```ts
type QueryParams = Record<string, unknown>

interface ListProps: {
  queryParams: QueryParams
}
```

Reply:
> there is no way to declare a type definition like that without the interface?

It can be inlined:
```ts
const List = ({ query: { selectedName } }: { query: { selectedName: string } }) => {
```

But if a component takes more than a few props, that gets very noisy.

Reply:
We create `selectedName` as the query param (and have it typed as a string), so we can enforce here that it's a string.

```
href: `${Nav.getLink('billing')}?${qs.stringify({ selectedName: projectName, type: 'project' })}`

Reply:
That's true when someone visits this page via that link. But a user could also delete the query string from the URL and load the page.

Reply:
Right, I need to allow it to be undefined as well.

Reply:
FWIW, the inline definition didn't work so I went with the interface defintion.